cmake/Platform/Mac/runtime_dependencies_mac.cmake.in

```diff
@@ -113,18 +113,26 @@ function(ly_copy source_file target_directory)
 
     endif()
 
-    cmake_path(COMPARE "${source_file}" EQUAL "${target_directory}/${target_filename}" same_location)
+    cmake_path(APPEND target_file "${target_directory}" "${target_filename}")
+    cmake_path(COMPARE "${source_file}" EQUAL "${target_file}" same_location)
     if(NOT ${same_location})
         if(NOT EXISTS "${target_directory}")
             file(MAKE_DIRECTORY "${target_directory}")
         endif()
-        if(NOT EXISTS "${target_directory}/${target_filename}" OR "${source_file}" IS_NEWER_THAN "${target_directory}/${target_filename}")
+
+        if(NOT target_is_bundle)
+            # if it is a bundle, there is no contention about the files in the destination, each bundle target will copy everything
+            # we dont want these files to invalidate the bundle and cause a new signature
+            file(LOCK ${target_file}.lock GUARD FUNCTION TIMEOUT 300)
+        endif()
+        file(SIZE "${source_file}" source_file_size)
+        if(EXISTS "${target_file}")
+            file(SIZE "${target_file}" target_file_size)
+        else()
+            set(target_file_size 0)
+        endif()
+        if((NOT source_file_size EQUAL target_file_size) OR "${source_file}" IS_NEWER_THAN "${target_directory}/${target_filename}")
```

Comment:
Same here.

Same with the `file(TOUCH_NOCREATE ${target_directory}/${target_filename})` below.